### ISSUE TITLE
notify.alexa_media_last_called_*****_hotmail_com does not always exist

### ISSUE THREAD

**Issue by geofffarmer** [alandtse/alexa_media_player#2362]
**Describe the bug**

This integration creates a service in my HA called notify.alexa_media_last_called_****_hotmail_com that I can use to get the last called echo to speak. However, this service does not always exist. This causes my automation that uses the service to get repeatedly flagged as a problem when the service doesn't exist. Can this be fixed?

**To Reproduce**

I'm not sure

**Expected behavior**

The service always exists

**System details**

- 2024.7 latest
- alexa_media (version from `const.py` or HA startup): 4.11.5
- alexapy (version from `pip show alexapy` or HA startup):
- Amazon 2FA: Yes

**Comment by danielbrunt57** [comment]
> * alexa_media (version from `const.py` or HA startup): 4.11.5

Is this still an issue with v4.12.1?

**Comment by geofffarmer** [comment]
Hi @danielbrunt57 

This is still an issue in 4.12.1.

Thank you for looking at this.

**Comment by danielbrunt57** [comment]
Can you please try v4.12.5?

**Comment by geofffarmer** [comment]
Hi @danielbrunt57 

I'm sorry to say that it is not fixed by 4.12.5

Do you need any more info?

Thanks for trying. Good luck!

**Comment by BoGnY** [comment]
same problem.. captcha error

![image](https://github.com/user-attachments/assets/1bf6e538-b8b1-41c5-bc28-056bd348a413)

```log
Registratore: homeassistant.config_entries
Fonte: config_entries.py:604
Prima occorrenza: 19:54:47 (1 occorrenze)
Ultima registrazione: 19:54:47

Error setting up entry XXXXXXXXXXXXXXXXX - amazon.it for alexa_media
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 604, in async_setup
    result = await component.async_setup_entry(hass, self)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/alexa_media/__init__.py", line 359, in async_setup_entry
    await login.login(cookies=await login.load_cookie())
  File "/usr/local/lib/python3.12/site-packages/alexapy/helpers.py", line 137, in wrapper
    return await func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/alexapy/alexalogin.py", line 687, in login
    post_resp = await self._session.post(
                ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/aiohttp/client.py", line 576, in _request
    raise err_exc_cls(url)
aiohttp.client_exceptions.InvalidUrlClientError: /errors/validateCaptcha
```

**Comment by ScottWoolven5** [comment]
I'm getting the same issue, no longer able to send notifications via my Alexa devices with my automation logging that notify is an unknown action. 

I get the same error as above and also a warning:
`This error originated from a custom integration.

Logger: alexapy.helpers
Source: custom_components/alexa_media/__init__.py:359
integration: Alexa Media Player (documentation, issues)
First occurred: 8 August 2024 at 18:20:49 (2 occurrences)
Last logged: 21:25:35

alexalogin.login((<alexapy.alexalogin.AlexaLogin object at 0x7f63741670>,), {'cookies': {'session-id': <Morsel: session-id=259-8453619-8200156; Path=/>, 'session-id-time': <Morsel: session-id-time=2082787201l; Path=/>, 'csrf': <Morsel: csrf=-1716822703; Path=/>, 'ubid-acbuk': <Morsel: ubid-acbuk=257-4776270-6295115; Path=/>, 'session-token': <Morsel: session-token=ah86uVxnc1OCtVrFuHkL9rk6UQisHvCiQ/PAFzpLMAY8bneYo2s9MhpTdseTmSx5Q6Ifd7muFlh7ukBr00vZTew1ZLIs36fr8sTUOafc0YZSrTHYWc3DCjyiZG9eQavKh4UBT/knqIQ8TRnzLWQuBVaTvTABvZ/1KQ48QEF92mLnR/KsSx8Yi0OtFFOlzul/gAaaZ1KNW/3q0Z6ntjlza9vNJSvYF8+5xJwwOVb7E3MAG9dCUjdIGUpHUO+FxVUJXLnIEylC2rrghRvJ63lthf5o5mItIrubdy9kBmZLbOxeFtbM41Wfkern1XvwrhsipjA2is/QO6YANA7T5T0Qtp6zgla/1RfftgHsVG223K5/ihLsd6cyJh5F/yNAoa1Y; Path=/>, 'x-acbuk': <Morsel: x-acbuk="ZSEzhOl@TaZwjriDwrY57KOcul2mX3tX"; Path=/>, 'i18n-prefs': <Morsel: i18n-prefs=GBP; Path=/>, 'lc-acbuk': <Morsel: lc-acbuk=en_GB; Path=/>}}): An error occurred accessing AlexaAPI: An exception of type InvalidUrlClientError occurred. Arguments: (URL('/errors/validateCaptcha'),)
alexalogin.login((<alexapy.alexalogin.AlexaLogin object at 0x7f63741670>,), {'cookies': {}}): An error occurred accessing AlexaAPI: An exception of type InvalidUrlClientError occurred. Arguments: (URL('/errors/validateCaptcha'),)`

**Comment by andy-81** [comment]
I am getting the same error with 4.12.7. 

I have a script which updates the last called details but I am getting an error when I try to send to the last called. I have an automation which uses notify.alexa_media_last_called which no longer appears in the actions/services tab of home assistant and throws up an error when the script runs.

**Comment by BoGnY** [comment]
for me is resolved with 4.12.7..
many others have to delete pickle file in .storage directory, many other have to reconfigure without using amazon.com as url but using their local Amazon site (ex amazon.it, amazon.de, etc)

**Comment by andy-81** [comment]
Thanks BoGnY,

deleting the pickle file appears to have done the job for me. Apologies if this was listed somewhere else but perfect job helping me out :)
